docs/resource-validation.md

```diff
@@ -0,0 +1,137 @@
+with the following error:
+
+```
+kubectl apply -f coffee-route.yaml 
+The HTTPRoute "coffee" is invalid: spec.hostnames[0]: Invalid value: "cafe.!@#$%example.com": spec.hostnames[0] in body should match '^(\*\.)?[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$'
+```
+
+> While unlikely, this validation step can be bypassed if the Gateway API CRDs are modified to remove the validation.
+> If this happens, Step 4 will ensure that invalid values (from NGINX perspective) are rejected.
+
+### Step 2 - Webhook Validation by Gateway API Webhook
+
+The Gateway API comes with a validating webhook which is enabled by default in the Gateway API installation manifests.
+It validates Gateway API resources using advanced rules not available in the OpenAPI schema validation. For example, if
+you create a Gateway resource with a TCP listener that configures a hostname, the webhook will reject it with the
+following error:
+
+```
+kubectl apply -f gateway.yaml 
+Error from server: error when creating "gateway.yaml": admission webhook "validate.gateway.networking.k8s.io" denied the request: spec.listeners[1].hostname: Forbidden: should be empty for protocol TCP
+```
+
+> This validation step can be bypassed if the webhook is not running in the cluster.
+> If this happens, Step 3 will ensure invalid values are rejected.
+
+### Step 3 - Webhook validation by NKG
+
+The previous step relies on the Gateway API webhook running in the cluster. To ensure that the resources are validated
+with the webhook validation rules even if the webhook is not running, NKG performs the same validation. However, NKG
+will perform the validation *after* a resource is already accepted by the Kubernetes API server.
+
+Below is an example of how NKG rejects an invalid resource (a Gateway resource with a TCP listener that configures a
+hostname) with a Kubernetes event:
+
+```
+kubectl describe gateway gateway
+. . .
+Events:
+  Type     Reason    Age   From                            Message
+  ----     ------    ----  ----                            -------
+  Warning  Rejected  6s    nginx-kubernetes-gateway-nginx  the resource failed webhook validation, however the Gateway API webhook failed to reject it with the error; make sure the webhook is installed and running correctly; validation error: spec.listeners[1].hostname: Forbidden: should be empty for protocol TCP; NKG will delete any existing NGINX configuration that corresponds to the resource
+```
+
+> This validation step always runs and cannot be bypassed.
+
+> NKG will ignore any resources that fail the webhook validation like in the example above.
+> If the resource previously existed, NKG will remove any existing NGINX configuration for that resource.
+
+### Step 4 - Validation by NKG
+
+This step catches the following cases of invalid values:
+
+* Values valid from the Gateway API perspective but not supported by NKG yet. For example, a certain filter in an
```

Comment:
Do we need an example here? Once we support all filters, will we have to update this doc?

Reply:
also added a link to the Gateway API compatibility doc